### DISCUSSION TITLE
Troubleshooting HAA Configuration for a Pool Controller - Sonoff 4ch

### DISCUSSION THREAD

**Discussion by MarcinRawski** [RavenSystem/esp-homekit-devices#2071] [MEPLHAA Scripts]
I am attempting to use HAA to configure a custom pool controller setup using an sonoff 4ch pro device. The device is meant to control four different relays corresponding to various pool functions.
 The GPIOs are configured as follows:

GPIO 15 garden lamp.
GPIO 12 pool pump.
GPIO 4 pool lamp.
GPIO 5 - electric heater.
GPIO 2 - DS18B20
GPIO14 - water flow sensor (pressure switch)

heater must always start the pump .

All individual functions are working correctly, except I'm unable to link the relay connected to GPIO5 with the relay connected to GPIO12. I would like the relay connected to GPIO5 to switch on whenever the relay connected to GPIO12 is activated.

Here is the HAA JSON configuration I'm using: 
```
{
  "c": {
    "io": [
      [[13],3],
      [[2], 4],
      [[14], 6],
      [[0, 9, 10],6],
      [[4, 5, 12, 15],2]],
    "pt": 8,
    "w": 10,
    "n": "Pool Controller,",
    "l": 13,
    "i": 1,
    "b": [
      {
        "g": 0,
        "t": 5
      }
    ],
    "m": 240
  },
   "a": [
    {
      "0": {
        "r": [
          {
            "g": 15
          }
        ]
      },
      "1": {
        "r": [
          {
            "g": 15,
            "v": 1
          }
        ]
      },
      "b": [
        {
          "g": 10,
          "t": 3
        }
      ],
      "s": 0
    },
    {
      "0": {
        "r": [
          {
            "g": 12,
            "v": 0
          }
        ]
      },
      "1": {
        "r": [
          {
            "g": 12,
            "v": 1
          }
        ]
      },
      "b": [
        {
          "g": 0,
          "t": 1
        }
      ],
      "s": 0,
      "f1": [],
      "f0": [
        {
          "g": 14,
          "t": 1,
          "i": 0
        },
        {
          "g": 5,
          "t": 0
        }
      ]
    },
    {
      "0": {
        "r": [
          {
            "g": 4
          }
        ]
      },
      "1": {
        "r": [
          {
            "g": 4,
            "v": 1
          }
        ]
      },
      "b": [
        {
          "g": 10
        }
      ],
      "s": 0,
      "i": 3600
    },
    {
      "0": {
        "r": [
          {
            "g": 5,
            "v": 0
          }
        ]
      },
      "3": {
        "r": [
          {
            "g": 5,
            "v": 1
          }
        ]
      },
      "t": 21,
      "b": [
        {
          "g": 9,
          "t": 1
        }
      ],
      "g": 2,
      "n": 3,
      "j": 90,
      "w": 1,
      "m": 20,
      "x": 35,
      "s": 0,
      "f0": [],
      "f1": []
    },
    {
      "t": 5,
      "s": 0,
      "f0": [
        {
          "g": 14,
          "t": 0
        }
      ],
      "f1": [
        {
          "g": 14,
          "t": 1
        }
      ]
    }
  ]
}

```
Despite my efforts, I haven't been successful in establishing this link between the two relays. Can anyone guide me on how to correctly configure this in the JSON setup? Has anyone encountered a similar issue or have any insights on how to resolve this?

Thank you in advance for your assistance.

**Comment by PhaseAngle** [comment]
put your live wire onto your normally opens put your pump and heater on their respective commons and link the heaters common to the pumps normally closed

**Comment by xrust83** [comment]
https://github.com/RavenSystem/esp-homekit-devices/wiki/accessory-configuration#service-notifications

**Comment by xrust83** [comment]
```
"1": {
        "r": [
          {
            "g": 12,
            "v": 1
          }
        ],
        "m": [
          [
            3, // - haa acsessory number - //
            1
          ]
        ]
      },
```

**Reply by PhaseAngle** [reply]
That is good on the software side but what happens if the pump is turned off while the heater is on.
This can happen because the pump is usually on a timer.
My own system has seperate pumps and is solar heated.
Probably best to do both

**Comment by gepardson** [comment]
@MarcinRawski
Hello,
Were you successful in making this work as you intended?
I was trying to use a Sonoff 4CH for the same purpose. If you managed to get this work the way you wanted, would you mind sharing the final script?
Thanks in advance.
